### DISCUSSION TITLE
Highlight Integration Error: not authorized to patch this environment variable

### DISCUSSION THREAD

**Discussion by MichaelCGrant** [vercel/community#2551] [Help] [answered]
### Summary

I am trying to setup the integration between my Vercel deployment and our Highlight.io monitoring. I am an Owner on the project in Vercel and an Admin on Highlight. When I get to the step to link the actual projects in each service, I get the following error:

Error: Vercel Project Env API responded with error; status_code=403 Forbidden; body={"error":{"code":"forbidden","message":"Not authorized to patch this environment variable."}}

![image](https://github.com/vercel/community/assets/93897555/a48a8b0b-a416-4d27-b46a-8b5a57345f2e)

Afterwards, the integration shows as enabled on both Highlight and Vercel. The sourcemap environment variable does get added to Vercel and it is correct. The log drain to Highlight does not get setup, however.

What can I do to enable this integration to complete successfully?

**Comment by amyegan** [comment]
Hi @MichaelCGrant. I wasn't able to reproduce the problem with my test app. If you can share the project id, then I can use that to search for more details.

**Reply by jay-khatri** [reply]
Hey @MichaelCGrant, we'll take a look on the Highlight end as well. Will report back with any findings.

**Reply by MichaelCGrant** [reply]
Sorry for the late reply.
Project ID is prj_uMbTdIouSZImh6KreMy6RlvNJDPI

**Reply by jay-khatri** [reply]
@amyegan I believe this is solved now. cc @MichaelCGrant

**Comment by deltaepsilon** [comment]
Starting the investigation.

First stop: `backend/vercel/vercel.go:SetEnvVar` in the Highlight repo.

https://github.com/highlight/highlight/blob/54768959f7c2404a1c809192334642c2c3ea1fc2/backend/vercel/vercel.go#L84

**Reply by deltaepsilon** [reply]
Appears to also involve `backend/private-graph/graph/schema.resolvers.go:UpdateVercelProjectMappings`: 

https://github.com/highlight/highlight/blob/54768959f7c2404a1c809192334642c2c3ea1fc2/backend/private-graph/graph/schema.resolvers.go#L3377

![image](https://github.com/vercel/community/assets/878947/a2fbfdf3-a2ca-4e7c-a86f-3796fcabf820)

Here's the feature PR: 

https://github.com/highlight/highlight/pull/3112

**Reply by mayberryzane** [reply, marked as answer]
Hi @MichaelCGrant - would you be able to delete the Highlight env variables from your Vercel project and set up the Highlight integration again? Failing to patch the env variables leads me to think those env variables are no longer associated with Highlight (e.g. if the Highlight integration was previously set up and then removed)

**Reply by MichaelCGrant** [reply]
Hi @mayberryzane, I removed the Highlight env variable and removed the integration from both sides (Highlight and Vercel). When I tried to connect the integration again, I got the same error.

**Reply by mayberryzane** [reply]
Hi @MichaelCGrant , apologies for the delayed response here. Just looked a bit closer into this issue, there are two environment variables - `NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID` and `HIGHLIGHT_SOURCEMAP_UPLOAD_API_KEY` - deleting both of these should resolve the issue since it happens when one of these env vars is already defined in the Vercel project and the Highlight integration does not have permissions to update it. Did you manually add either as env variables before installing the integration? I think one of our Next.js docs pages instructs users to define `NEXT_PUBLIC_HIGHLIGHT_PROJECT_ID`, which could be causing this error if it's done manually before the Highlight integration is installed. We'll update our docs and the error messaging to make this more clear.

**Reply by MichaelCGrant** [reply]
That did the trick! Thanks! I guess I missed deleting the NEXT_PUBLIC_ variable.. Thanks again!
